### PR TITLE
Edit username

### PR DESCRIPTION
**Pull request by TolulopeJoel**
## Description

This commits adds modifications to the `google_callback` function, which now generates a unique username based on the user's display name if the user is not already in the database.
And also added and edit_username endpoint that make users edit their username.

## Motivation and Context
Update to the user registration and authentication process will provide a more streamlined and user-friendly experience. This change efficiently handle the way users authenticating with google, ensuring that they can easily access the app.

## How Has This Been Tested?
Tested this code with different scenarios, such as registering new users and checking for conflicts with existing users while trying to sign up with google with a display name that already exists.

## Screenshots (if appropriate - Postman, etc):

**Create new user using email and password**
Screenshot 2023-10-16 at 10 12 55 AM
Screenshot 2023-10-16 at 11 07 55 AM

**Try to sign up to google with a username that already exists in database**
Screenshot 2023-10-16 at 10 13 16 AM

**Success message with username uniqueness handled well**
Screenshot 2023-10-16 at 10 14 20 AM
Screenshot 2023-10-16 at 11 08 38 AM

**Try to edit username with already existing username**
Screenshot 2023-10-16 at 11 09 56 AM

**Succes updating username**
Screenshot 2023-10-16 at 11 10 31 AM
Screenshot 2023-10-16 at 11 11 02 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Check the relevant box to indicate the type of change. In your case, this is a new feature.

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: app/routes/auth_routes.py [hunk 1]
```diff
@@ -266,5 +287,31 @@ async def google_callback(
     return UserResponse(
         status_code=200,
         message="User Logged in Successfully!",
-        username=user_display_name,
+        username=new_user.username,
     )
+
+
+@auth_router.put("/edit_username/{user_id}/")
```
Comment (zxenonx) [inline app/routes/auth_routes.py:294]: The http method is PUT, so we know the endpoint will modify a resource.
Then the route could just be
/username/{user_id}/

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @TolulopeJoel 
I left a comment

File: app/routes/auth_routes.py (hunk 1 above)
Comment (TolulopeJoel) [reply app/routes/auth_routes.py]: I have changed the endpoint URL to what's suggested

**Review by zxenonx — APPROVED** [review]
Good.

**Pull request merged**